[BE-104] Fix pipeline examples

rossgray opened the pull request:
Currently, quite a few of the examples are failing. This PR fixes these. I'm not too familiar with all the expected behaviour of this library so any help here would be greatly appreciated.

rossgray commented on examples/api_pipeline_run.py:
```diff
@@ -22,8 +20,6 @@ def add_lol(a: str) -> str:
-except (TypeError, json.JSONDecodeError):
-    result_preview = "unavailable"
+
+result_preview = run_result["result_preview"]
```
result preview is a list so JSON decoding was failing

alexpearce replied:
It might be that the `try`/`except` handles the case of a failed run. The Run API doesn't offer a very clear contract in this case at the moment.

rossgray replied:
Yeah I'm not sure to be honest. Looking at the [type annotation](https://github.com/mystic-ai/pipeline/blob/main/pipeline/schemas/run.py#L92) for the `result_preview` it is supposed to be a `list` or `dict` and `json.loads` expects a `str` so this would never work. Perhaps we were trying to serialise the `dict` into a string instead? In any case, I will ignore the JSON stuff and add a `try/except` to catch the case where there is no `result_preview`

alexpearce commented on examples/basic_maths/README.md:
````diff
@@ -74,5 +79,4 @@ with Pipeline("MathsIsFun") as pipeline:
 output_pipeline = Pipeline.get_pipeline("MathsIsFun")
 print(output_pipeline.run(5.0, 6.0))
-
 ```
````
Well done for remembering to check the README 😆 so easy to miss.

alexpearce commented on examples/huggingface/gpt-j-6b.py:
```diff
@@ -130,6 +130,7 @@ def __exit__(self, type, value, traceback):
                 self.tokenizer = AutoTokenizer.from_pretrained(self.tokenizer_path)
 
 
+# NOTE: requires PIPELINE_API_TOKEN environment variable to be set
 api = PipelineCloud()
```
Alternatively we could pass in an obvious placeholder token (no strong preference from me).

```suggestion
# Replace this token with your own
api = PipelineCloud(token="pipeline_token_value")
```

rossgray replied:
Thanks. I've added your suggestion but also noted that they can alternatively use the env var

alexpearce reviewed (commented):
Thanks very much for tidying things up!

I guess some (all?) of these require access to the API, otherwise it would be nice to automate the testing of these to avoid them drifting away from the spec in the future.

rossgray commented:
Thanks @alexpearce for reviewing. I've done a bit more cleanup:
- I've deleted the old examples that we don't want to support currently
- I've combined `examples/api_usage.py` and `examples/api_pipeline_run.py` and deleted the latter, since there was a lot of overlap between the two and I think just having one clear example is better
- I've applied some changes based on your feedback (API tokens + `try/except` handling)

> I guess some (all?) of these require access to the API, otherwise it would be nice to automate the testing of these to avoid them drifting away from the spec in the future.
Most, but not all, required API access. We could still add some tests by mocking parts of the API client. Let me know if you'd like me to do this or not.

alexpearce reviewed (approved):
👏

rossgray commented:
sorry @alexpearce, I needed to bump the version in `pyproject.toml`
